.github/workflows/workflow.yml

```diff
@@ -21,9 +21,56 @@ jobs:
     - name: Run Tox Check
       run: tox -e check
 
-  build:
+  data-cache:
     needs: check
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e ".[dev]"
+      - name: Restore Cache
+        uses: actions/cache@v4
+        with:
+          key: cached-data-${{ github.ref_name }}
+          restore-keys: |
+            cached-data-develop
+            cached-data-main
+          path: .github_cache/ASPIRE-data
+          enableCrossOsArchive: true
+      - name: Create Cache Directory
+        run: |
+          # Create cache and config directories
+          mkdir -p .github_cache/ASPIRE-data
+          chmod -R 777 .github_cache/ASPIRE-data
+          echo "cache:" > config.yaml
+          echo "    cache_dir: .github_cache/ASPIRE-data" >> config.yaml
+          echo "Logging config.yaml for verification:"
+          cat config.yaml
+      - name: Download EMDB 2660
+        run: |
+          export ASPIREDIR=.
+          python -c "from aspire.downloader import emdb_2660; emdb_2660()"
+      - name: Verify Cache Directory Before Saving
+        run: find . -name config.yaml
```

Comment:
If this is meant to be a check it can just be automated by changing the call to exit on failure.

Reply:
will do.

Reply:
Just want to update that I changed this to use `ls` so that the actual cached files are listed in the CI output for quick visual inspection.

Reply:
cool, I like the idea of the `ls` listing for potential debugging, but I think having an automated check is still relevant before moving on to later jobs.  You can use `exec` or `print` feature of `find`, or maybe use github's CI stuff if that works.

```
  artifact:
    paths:
```

Reply:
ok, using `ls` and have the automated check now.